cvat/rqworker.py

```diff
@@ -1,16 +1,34 @@
 # Copyright (C) 2018-2022 Intel Corporation
-# Copyright (C) 2022-2023 CVAT.ai Corporation
+# Copyright (C) 2022-2024 CVAT.ai Corporation
 #
 # SPDX-License-Identifier: MIT
 
 import os
-
+import signal
 from rq import Worker
+from rq.worker import StopRequested
 
 import cvat.utils.remote_debugger as debug
 
+class CVATWorker(Worker):
+    # may be called from work horse's perform_job::except block
+    # or from parent's Worker::monitor_work_horse_process
+    # if parent process sees that work-horse is dead
+
+    # This modification ensures that jobs stopped intentionally
+    # do not get their status updated or placed in the failed registry
+    # as the main server code must delete them at all
+    def handle_job_failure(self, job, queue, **kwargs):
+        # pylint: disable=access-member-before-definition
+        if self._stopped_job_id == job.id:
+            self._stopped_job_id = None
+            self.set_current_job_id(None)
+        else:
+            super().handle_job_failure(job, queue, **kwargs)
```

Comment:
I see one problem here - a started rq job can also be stopped from django admin panel and with these changes such a job won't be handled correctly.

Reply:
> won't be handled correctly

Do you have ideas why? If it sends kill signal to the work horse process and does not set `self._stopped_job_id`, it is probably problem with rq.

Finally, this fix does not suppose to solve all known problems with rq jobs, so:

- the problem was not added with this patch, and
- you do not see any problems in regular user scenarios, 
 
I do not think it will block us from merging.